app/javascript/components/zone-form/index.jsx

```diff
@@ -0,0 +1,29 @@
+import React, { useEffect, useState } from 'react';
+
+import MiqFormRenderer from '@@ddf';
+import createSchema from './zone-form.schema';
+
+const ZoneForm = ({recordId}) => {
+    const [initialValues, setInitialValues] = useState([]);
+    console.log(recordId);
+
+    useEffect(() => {
+        if (recordId) {
+            API.get(`/api/zones/${recordId}`).then((res)=> {
+                console.log(res);
+                setInitialValues(res);
```

Comment:
This side effect can cause visual glitches on the page before the values are actually loaded. If the network is slow or the user is fast enough, it is possible to edit the form before the data gets loaded. This can cause undeterministic behavior as the `initialValues` can be set on the form only on its first render.

In order to avoid this, you could introduce an `isLoading` attribute in your state that would have a `true` initial value if the `recordId` is set.
```js
const [{ initialValues, isLoading }, setState] = useState({ isLoading: !!recordId })
```

After the data is loaded from the API, you should both set the `initialValues` and the `isLoading` to `false`:
```
setState({ initialValues, isLoading: false });
```

And finally make the rendering of the form conditional:
```
return !isLoading && (
  <MiqFormRenderer ...
```

If you set up everything correctly, the delayed render should only happen when the form is in edit mode.